wallet/src/de/schildbach/wallet/ui/username/voting/ConfirmUsernameRequestDialogFragment.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.ui.username.voting
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import dagger.hilt.android.AndroidEntryPoint
+import de.schildbach.wallet_test.R
+import de.schildbach.wallet_test.databinding.DialogConfirmUsernameRequestBinding
+import org.dash.wallet.common.ui.dialogs.OffsetDialogFragment
+import org.dash.wallet.common.ui.viewBinding
+import org.dash.wallet.common.util.observe
+
+@AndroidEntryPoint
+class ConfirmUsernameRequestDialogFragment: OffsetDialogFragment(R.layout.dialog_confirm_username_request) {
+    private val binding by viewBinding(DialogConfirmUsernameRequestBinding::bind)
+
+    private val viewModel by viewModels<ConfirmUserNameDialogViewModel>()
+    private val requestUserNameViewModel by activityViewModels<RequestUserNameViewModel>()
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.confirmBtn.setOnClickListener {
+            requestUserNameViewModel.submit()
+
+            dismiss()
+        }
+
+        binding.dismissBtn.setOnClickListener { dismiss() }
+
+        viewModel.exchangeRate.observe(viewLifecycleOwner) { viewModel.updateFees(it) }
```

Comment:
Why observe `exchangeRate` from the viewModel in the fragment, just to update fees in the same viewModel? All of this can be moved into the viewModel.

Reply:
Refactored